spec/lib/kafka/schema_registry/service_spec.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'kafka/schema_registry/service'
+
+describe Kafka::SchemaRegistry::Service do
+  let(:service) { described_class.new }
+  let(:subject_name) { 'topic-1' }
+  let(:version) { '1' }
+  let(:topic_1_response) do
+    { 'subject' => 'topic-1-value',
+      'version' => 1,
+      'id' => 5,
+      'schema' => "{\"type\":\"record\",\"name\":\"TestRecord\",\"namespace\":\"gov.va.eventbus.test.data\"\
+,\"fields\":[{\"name\":\"data\",\"type\":{\"type\":\"map\",\"values\":\"string\"},\"default\":{}}]}" }
+  end
+
+  describe '#subject_version' do
+    context 'when requesting latest version' do
+      it 'returns schema information' do
+        VCR.use_cassette('kafka/topics') do
+          response = service.subject_version(subject_name)
+          expect(response).to eq(topic_1_response)
+        end
+      end
+    end
+
+    context 'when requesting specific version' do
+      it 'returns schema information for that version' do
+        VCR.use_cassette('kafka/topics') do
+          response = service.subject_version(subject_name, version)
+          expect(response).to eq(topic_1_response)
+        end
+      end
+    end
+
+    context 'when subject does not exist' do
+      let(:nonexistent_subject) { 'topic-999' }
+
+      it 'raises a ResourceNotFound error' do
+        VCR.use_cassette('kafka/topics404') do
+          expect { service.subject_version(nonexistent_subject) }
+            .to raise_error(Faraday::ResourceNotFound)
+        end
+      end
+    end
+  end
+
+  describe 'request handling' do
+    context 'when making a GET request' do
+      it 'receives expected 200 response' do
+        VCR.use_cassette('kafka/topics') do
+          response = service.send(:get, "/subjects/#{subject_name}-value/versions/#{version}")
+          expect(response).to eq(topic_1_response)
+        end
+      end
+    end
+
+    context 'when response is not valid JSON' do
```

Comment:
I'm not sure how useful this spec is for our purposes. We're testing more the functionality of the parent class `Common::Client::Base`. Perhaps remove for the "request handling" specs for now and we can later add specs for the other schema registry API endpoints as we start using them.

Reply:
Right, that makes sense. Sounds good. Removed